packages/good-design/src/apps/onramp/OnramperController.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { Onramper } from "./Onramper";
+import { useEthers, useEtherBalance, useTokenBalance, useCall } from "@usedapp/core";
+import { WebViewMessageEvent } from "react-native-webview";
+import { useContractFunctionWithDefaultGasFees, useGetContract } from "@gooddollar/web3sdk-v2";
+import * as ethers from "ethers";
+import { Contract } from "ethers";
+import { useModal } from "../../hooks/useModal";
+import { View, Text } from "native-base";
+import { WalletAndChainGuard } from "../../core";
+import { useSignWalletModal } from "../../hooks/useSignWalletModal";
+
+export const OnramperController = () => {
+  const cusd = "0x765de816845861e75a25fca122bb6898b8b1282a";
+  const { account, chainId, library } = useEthers();
+  const swapLock = useRef(false);
+
+  const { SignWalletModal } = useSignWalletModal();
+
+  // const buygdFactory = useGetContract("BuyGDFactory", false, "base");
+  const buygdFactory = new Contract("0x00e533B7d6255D05b7f15034B1c989c21F51b91C", [
+    "function createAndSwap(address owner,uint256 minAmount) external returns(address)",
+    "function predict(address owner) external view returns(address)"
+  ]);
+
+  const targetGDHelper = useCall(
+    account &&
+      buygdFactory && {
+        contract: buygdFactory,
+        method: "predict",
+        args: [account]
+      },
+
+    { refresh: "never", chainId }
+  );
+  const gdHelperAddress = targetGDHelper?.value?.[0];
+  const buygdInstance = new Contract(gdHelperAddress || ethers.constants.AddressZero, [
+    "function swap(uint256 minAmount,address gasRefund) external"
+  ]);
+  const celoBalance = useEtherBalance(gdHelperAddress, { refresh: 1 });
+  // const accountCeloBalance = useEtherBalance(account, { refresh: 1 }) || ethers.constants.Zero;
+
+  const cusdBalance = useTokenBalance(cusd, gdHelperAddress, { refresh: 1 });
+
+  const { showModal, Modal } = useModal();
+  const { send: swap, state: swapState } = useContractFunctionWithDefaultGasFees(buygdInstance, "swap", {
+    transactionName: "Exchange bought tokens to G$"
+  });
+
+  const { send: createAndSwap, state: createState } = useContractFunctionWithDefaultGasFees(
+    buygdFactory,
+    "createAndSwap",
+    {
+      transactionName: "Deploy secure swap helper"
+    }
+  );
+
+  const selfSwap = false; //accountCeloBalance.gt(ethers.utils.parseUnits("5", 9).mul(400000)); //400000 gas for swap * 5gwei gas price
+
+  const [step, setStep] = useState(-1);
+
+  // console.log({ selfSwap, account, gdHelperAddress, accountCeloBalance, cusdBalance, celoBalance });
+  /**
+   * callback to get event from onramper iframe
+   */
+  const callback = useCallback((event: WebViewMessageEvent) => {
+    if ((event.nativeEvent.data as any).title === "success") {
```

Comment:
add a note in the issue or somewhere that this needs to be tested and verified

Reply:
done